tests/foreman/virtwho/test_cli_virtwho.py

```diff
@@ -375,3 +377,22 @@ def test_positive_rhsm_option(self):
             get_configure_option('rhsm_prefix', config_file), '/rhsm')
         VirtWhoConfig.delete({'name': name})
         self.assertFalse(VirtWhoConfig.exists(search=('name', name)))
+
+    @tier2
+    def test_positive_post_hypervisors(self):
+        """ Post large json file to /rhsm/hypervisors"
+
+        :id: e344c9d2-3538-4432-9a74-b025e9ef852d
+
+        :expectedresults:
+            json file can be posted normally and the task is success
+
+        :CaseLevel: Integration
+
+        :CaseImportance: Medium
+
+        :BZ: 1637042
+        """
+        filename = "/tmp/test.json"
+        hypervisor_json_create(filename, hypervisors=100, guests=10)
+        hypervisor_json_post(filename)
```

Comment:
This seems wasteful. You save data to file, send that file to remote host and then ssh to the same host to issue `curl` command.

How about generating dict with data, storing it in variable and using `requests.post()` to send it to remote host?

Reply:
Great idea, I will commit again.
I coped the json file to remote satellite because I'm trying to post it by satellite cert:
`curl -X POST -s -k --cert /etc/pki/consumer/cert.pem --key /etc/pki/consumer/key.pem`
but finally, I changed to post it by username and password.

Reply:
Thanks for implementing that, it looks much cleaner now.